### ISSUE TITLE
Solve the general case of deduplication

### ISSUE THREAD

**Issue by mollerhoj** [centre-for-humanities-computing/danish-foundation-models#11]

**Comment by mollerhoj** [comment]
```
# -*- coding: utf-8 -*-

from lsh import cache, minhash
import itertools

# The size of char_ngrams to use
char_ngram = 5

# The number of minhashes to use (use list of seeds for determinism)
seeds = 100

# The number of bands to divide fingerprints into
num_bands = 100

# How many of the bands should be hit before with mark a line as a duplicate
threshold = 90

# hashbytes sets number of bytes to represent minhashes.
# set this to 4 in case we're running on a 32-bit architecture (unlikely)
hasher = minhash.MinHasher(seeds, char_ngram, hashbytes=8)
lshcache = cache.Cache(hasher, num_bands)

with open('output.txt', 'w') as output_file:
    with open('input.txt', 'r') as input_file:
        doc_id = 0
        for line in input_file:
            duplicates = 0
            ascii_string = line.decode('ascii', errors="ignore")
            fingerprint = hasher.fingerprint(ascii_string)
            lshcache.fingerprints[doc_id] = fingerprint
            for bin_i, bucket in lshcache.bins_(fingerprint):
                bucket_id = hash(tuple(bucket))
                if len(lshcache.bins[bin_i][bucket_id]) == 0:
                    lshcache.bins[bin_i][bucket_id].add(doc_id)
                else:
                    duplicates += True

            if duplicates < threshold:
                output_file.write(line)
            else:
                print(duplicates, line[0:50])
            doc_id +=1
```

**Comment by KennethEnevoldsen** [comment]
Thanks, @mollerhoj I will add this to a script during the hackathon.

**Comment by saattrupdan** [comment]
Implemented in #44 and #61.
